File: JS/Tema4_Ejercicio8.js
/*
 * TEMA 4: ESTRUCTURAS REPETITIVAS - EJERCICIO 8
 * ==============================================
 * Juego de adivinanza con break.
 * 
 * EXPLICACIÓN:
 * - Break para salir del bucle cuando se adivina
 * - Contador de intentos
 * - Comparación de números
 */

function juegoAdivinanza(numeroSecreto = 42, maxIntentos = 7) {
    console.log("=== JUEGO DE ADIVINANZA ===");
    console.log(`Intenta adivinar un número entre 1 y 100.`);
    console.log(`Tienes hasta ${maxIntentos} intentos.\n`);
    
    // Simulación con números predefinidos
    const intentos = [50, 30, 40, 45, 42];
    
    for (let i = 0; i < Math.min(intentos.length, maxIntentos); i++) {
        const intento = intentos[i];
        console.log(`Intento ${i + 1}: ${intento}`);
        
        if (intento === numeroSecreto) {
            console.log(`¡CORRECTO! ¡Ganaste en ${i + 1} intentos!`);
            break;  // Sale del bucle
        } else if (intento < numeroSecreto) {
            console.log(`  → Es mayor`);
        } else {
            console.log(`  → Es menor`);
        }
        
        if (i === Math.min(intentos.length, maxIntentos) - 1) {
            console.log(`\n¡Perdiste! El número era: ${numeroSecreto}`);
        }
    }
}

juegoAdivinanza();

/*
 * SALIDA ESPERADA:
 * === JUEGO DE ADIVINANZA ===
 * Intenta adivinar un número entre 1 y 100.
 * Tienes hasta 7 intentos.
 * 
 * Intento 1: 50
 *   → Es menor
 * Intento 2: 30
 *   → Es mayor
 * Intento 3: 40
 *   → Es mayor
 * Intento 4: 45
 *   → Es menor
 * Intento 5: 42
 * ¡CORRECTO! ¡Ganaste en 5 intentos!
 */

File: Java/Tema4_Ejercicio8.java
/*
 * TEMA 4: ESTRUCTURAS REPETITIVAS - EJERCICIO 8
 * ===============================================
 * Escriba un programa que use break para terminar un bucle cuando el usuario 
 * ingrese un número específico.
 */

import java.util.Scanner;

public class Tema4_Ejercicio8 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.println("=== JUEGO: ADIVINA EL NÚMERO ===");
        System.out.println("Estoy pensando en un número entre 1 y 10.");
        System.out.println("Escribe 'salir' para abandonar.");
        
        int numeroSecreto = 7;
        boolean adivinado = false;
        int intentos = 0;
        
        while (true) {
            System.out.print("\nIntenta adivinar: ");
            String input = scanner.nextLine();
            
            if (input.equalsIgnoreCase("salir")) {
                System.out.println("¡Hasta luego! El número era: " + numeroSecreto);
                break;
            }
            
            try {
                int numero = Integer.parseInt(input);
                intentos++;
                
                if (numero == numeroSecreto) {
                    System.out.println("¡CORRECTO! Adivinaste en " + intentos + " intentos.");
                    adivinado = true;
                    break;
                } else if (numero < numeroSecreto) {
                    System.out.println("El número es mayor.");
                } else {
                    System.out.println("El número es menor.");
                }
            } catch (NumberFormatException e) {
                System.out.println("Ingresa un número válido.");
            }
        }
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA (ejemplo):
 * === JUEGO: ADIVINA EL NÚMERO ===
 * Estoy pensando en un número entre 1 y 10.
 * Escribe 'salir' para abandonar.
 * 
 * Intenta adivinar: 5
 * El número es mayor.
 * 
 * Intenta adivinar: 8
 * El número es menor.
 * 
 * Intenta adivinar: 7
 * ¡CORRECTOINT! Adivinaste en 3 intentos.
 */
